server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -2189,6 +2189,58 @@ protected void validateConditionsToReplaceDiskOfferingOfVolume(VolumeVO volume,
         s_logger.info(String.format("Changing disk offering to [uuid=%s] while migrating volume [uuid=%s, name=%s].", newDiskOffering.getUuid(), volume.getUuid(), volume.getName()));
     }
 
+    /**
+     *  Checks if the target storage supports the new disk offering.
+     *  This validation is consistent with the mechanism used to select a storage pool to deploy a volume when a virtual machine is deployed or when a new data disk is allocated.
+     *
+     *  The scenarios when this method returns true or false is presented in the following table.
+     *
+     *   <table border="1">
+     *      <tr>
+     *          <th>#</th><th>Disk offering tags</th><th>Storage tags</th><th>Does the storage support the disk offering?</th>
+     *      </tr>
+     *      <body>
+     *      <tr>
+     *          <td>1</td><td>A,B</td><td>A</td><td>NO</td>
+     *      </tr>
+     *      <tr>
+     *          <td>2</td><td>A,B,C</td><td>A,B,C,D,X</td><td>YES</td>
+     *      </tr>
+     *      <tr>
+     *          <td>3</td><td>A,B,C</td><td>X,Y,Z</td><td>NO</td>
+     *      </tr>
+     *      <tr>
+     *          <td>4</td><td>null</td><td>A,S,D</td><td>YES</td>
+     *      </tr>
+     *      <tr>
+     *          <td>5</td><td>A</td><td>null</td><td>NO</td>
+     *      </tr>
+     *      <tr>
+     *          <td>6</td><td>null</td><td>null</td><td>YES</td>
+     *      </tr>
+     *      </body>
+     *   </table>
+     */
+    protected boolean doesTargetStorageSupportNewDiskOffering(StoragePool destPool, DiskOfferingVO newDiskOffering) {
+        String newDiskOfferingTags = newDiskOffering.getTags();
+        return doesTargetStorageSupportDiskOffering(destPool, newDiskOfferingTags);
+    }
+
+    @Override
+    public boolean doesTargetStorageSupportDiskOffering(StoragePool destPool, String diskOfferingTags) {
+        if (org.apache.commons.lang.StringUtils.isBlank(diskOfferingTags)) {
+            return true;
+        }
+        String storagePoolTags = getStoragePoolTags(destPool);
+        if (org.apache.commons.lang.StringUtils.isBlank(storagePoolTags)) {
+            return false;
+        }
+        String[] storageTagsAsStringArray = org.apache.commons.lang.StringUtils.split(storagePoolTags, ",");
```

Comment:
there is a  StringUtils.csvTagsToList()

Reply:
Are you talking about `com.cloud.utils.StringUtils.csvTagsToList(String)`?
Didn't we talk about stop using that "utils" class, so we can then remove it in the future?

Reply:
I think we agreed to move all string operation in it so it we have a small interface plain to external libraries.

Reply:
Did we?! That means we would create a layer over a library. I think we had a discussion, but I was under the impression that the idea was not to redo the wheel. Of course, my memory is not as good as it was before, so I am probably mistaken.

Reply:
In my opinion it is best to isolate all external libraries in wrappers so porting from version to version or from vendor to vendor becomes easier/more isolated. In fact the hypervisor guru and resource and data motion strategies are examples of that as well. We didn't make an onmail list dicision of it but I thought aI had convinced you of it.

Reply:
Well, but those "gurus"/plugins/drivers and so on are different "things".
I mean, they are not library, they are part of our code base to deliver our business. On the other hand, creating a layer over third party utils libraries such as StringUtils, NumberUtils, Hibernate, Spring and so on sound like an overkill.

We should discuss that we beers in Montreal :)

Reply:
one word: gson
and a lot of beers